src/package-scanner.ts

```diff
@@ -111,6 +114,15 @@ export class PackageScanner {
    * Scan each document in a package manually.
    */
   scanPackageManually() {
+    // Scan top-level entrypoints first, to make sure thier dependencies are
+    // properly converted to JS modules as well.
```

Comment:
(from offline) Pull the logic that switches the scan result type based on `topLevelEntrypoints` out of `scanDocument` and only do it here.

Reply:
done